arch/arc/core/swap.S

```diff
@@ -162,11 +162,21 @@ _swap_return_from_firq:
 	lr r3, [_ARC_V2_STATUS32]
 	bbit1 r3, _ARC_V2_STATUS32_AE_BIT, _return_from_exc_irq
 
+#ifdef CONFIG_ARC_HAS_SECURE
+	lr r3, [_ARC_V2_SEC_STAT]
+	/* set SEC_STAT.IRM = SECURE for interrupt return */
+	bset r3, r3, 3
+	/* sflag r3 */
+	/* sflag instruction is not supported in current ARC GNU */
+	.long 0x00ff302f
+#endif
+
 	/* pretend interrupt happened to use rtie instruction */
```

Comment:
Move the "pretend ..." comment to before the added SEC_STAT IRM update, the IRM update is also for the fake rtie